applications/vessel-history/src/features/profile/components/activity/ActivityVoyage.tsx

```diff
@@ -22,35 +22,37 @@ const ActivityVoyage: React.FC<EventProps> = ({
 
   const voyageLabel = useMemo(() => {
     const parts: string[] = []
+    const { from, to, start, end, eventsQuantity } = event
 
-    if (event.from && event.to) {
+    if (from && to) {
       parts.push(
-        `${t('common.from', 'from')} ${formatI18nDate(event.start ?? 0, {
-          format: DateTime.DATETIME_FULL,
+        `${t('common.from', 'from')} ${formatI18nDate(start ?? 0, {
+          format: DateTime.DATE_FULL,
         })}`
       )
       parts.push(
-        `${t('common.to', 'to')} ${formatI18nDate(event.end ?? 0, {
-          format: DateTime.DATETIME_FULL,
+        `${t('common.to', 'to')} ${formatI18nDate(end ?? 0, {
+          format: DateTime.DATE_FULL,
         })}`
       )
+      parts.push(`(${eventsQuantity} ${t('common.events', 'Events')})`)
     } else if (!event.to) {
       parts.push(t('event.currentVoyage' as any, 'Ongoing Voyage') as string)
       parts.push(
-        `${t('common.from', 'from')} ${formatI18nDate(event.start ?? 0, {
-          format: DateTime.DATETIME_FULL,
+        `${t('common.from', 'from')} ${formatI18nDate(start ?? 0, {
+          format: DateTime.DATE_FULL,
         })}`
       )
     } else {
       parts.push(
-        `${t('common.to', 'to')} ${formatI18nDate(event.end ?? 0, {
-          format: DateTime.DATETIME_FULL,
+        `${t('common.to', 'to')} ${formatI18nDate(end ?? 0, {
+          format: DateTime.DATE_FULL,
         })}`
       )
     }
 
-    return parts.join(' - ')
-  }, [event.from, event.start, event.to, event.end, t])
+    return parts.join(' ')
+  }, [event, t])
```

Comment:
If you leave the complete `event` as a dependency instead of enumerating every event property as it was before it will force to recompute the `voyageLabel` value on any change to the `event` (performance degradation). 

```suggestion
  }, [event.from, event.start, event.to, event.end, event.eventsQuantity, t])
```

Reply:
Looks like it computes the same amount of times in this case. Maybe because the other properties inside `event` don't change. But thanks for the heads up! I'll try to be mindful in other instances :)